src/transformers/models/encoder_decoder/modeling_encoder_decoder.py

```diff
@@ -372,6 +372,8 @@ def from_encoder_decoder_pretrained(
 
             if "config" not in kwargs_encoder:
                 encoder_config = AutoConfig.from_pretrained(encoder_pretrained_model_name_or_path)
```

Comment:
```suggestion
                encoder_config, kwargs_encoder = AutoConfig.from_pretrained(encoder_pretrained_model_name_or_path, **kwargs_encoder, return_unused_kwargs=True)
```

Reply:
I think this is actually the cleanest way to fix the problem. This way everything of `kwargs_encoder` that is relevant for the config is removed and everything that is relevant for the model is kept, *e.g.* `local_files_only=True`. This way we can still do:

```python
AutoModel.from_pretrained(..., **kwargs_encoder)
```

later.